### DISCUSSION TITLE
Get async session outside of route

### DISCUSSION THREAD

**Discussion by breno-aumo** [litestar-org/advanced-alchemy#161] [Q&A] [answered]
I followed the documentation (https://github.com/jolt-org/advanced-alchemy/blob/main/examples/fastapi.py) for building an async app.
```
alchemy: StarletteAdvancedAlchemy
alchemy = StarletteAdvancedAlchemy(config=sqlalchemy_config, app=app)
async def provide_db_session(request: Request) -> AsyncSession:
    """Provide a DB session."""
    return alchemy.get_session(request)
```

This allows me to get access to the database in my routes using with `Depends(provide_db_session)`.

For testing purposes, I want to be able to get a session outside of my routes. I should get an async session, perform some testing creations and deletions and exit. How can I get a session without a `Request` object?

**Comment by abnerjacobsen** [comment]
I'm also interested in that. Can anyone share an example of how to do it?

Thanks.

**Comment by cofin** [comment, marked as answer]
Sorry for the delay in answering this.  There is actually a method on the `config` object to [get a new session](https://github.com/litestar-org/advanced-alchemy/blob/main/advanced_alchemy/config/asyncio.py#L74) anywhere outside of a request.

you can use it like so:
```python
alchemy: StarletteAdvancedAlchemy
alchemy = StarletteAdvancedAlchemy(config=sqlalchemy_config, app=app)
async with sqlalchemy_config.get_session() as db_session:
     await db_session.execute(...)

```

If you needed to use this in a test,  you would just use this a fixture or monkey patch the session wherever you need it.

**Reply by cofin** [reply]
[This reference](https://github.com/litestar-org/litestar-fullstack/blob/main/tests/integration/conftest.py) from the Litestar fullstack test suite may be useful as well 

It creates an engine, session maker, and session fixture.  It also shows how to monkey patch the fixtures into your application (`_patch_db`)
